src/mlpack/methods/ann/layer/softshrink_impl.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file softshrink.hpp
+ * @author Lakshya Ojha
+ *
+ * Implementation of Soft Shrink activation function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_IMPL_HPP
+
+// In case it hasn't yet been included
+#include "softshrink.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+// This constructor is called for Soft Shrink activation function.
+// lambda is a hyperparameter.
+template<typename InputDataType, typename OutputDataType>
+SoftShrink<InputDataType, OutputDataType>::SoftShrink(const double lambda) :
+    lambda(lambda)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void SoftShrink<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, OutputType&& output)
+{
+  output = (input > lambda) % (input - lambda) + (
+            input < -lambda) % (input + lambda);
```

Comment:
Should be tabbed twice (4 spaces).

Reply:
Thanks for noticing.
Applied in the new changes.